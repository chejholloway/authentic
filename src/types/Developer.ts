export default interface Developer {
  avatar_url: string;
  login: string;
}
