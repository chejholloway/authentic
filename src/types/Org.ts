/**
 * Represents a GitHub organization's information.
 *
 * This interface defines the structure of information associated with a GitHub organization.
 * It includes details such as the organization's name, profile URL, avatar URL, and more.
 *
 * @interface Org
 * @example
 * ```ts
 * const organization: Org = {
 *   login: "exampleOrg",
 *   id: 789,
 *   avatar_url: "https://example.com/org-avatar.jpg",
 *   // ...other properties
 * };
 * ```
 */
export default interface Org {
  avatar_url: string;
  description: string;
  events_url: string;
  hooks_url: string;
  id: number;
  issues_url: string;
  login: string;
  members_url: string;
  node_id: string;
  public_members_url: string;
  repos_url: string;
  url: string;
}
