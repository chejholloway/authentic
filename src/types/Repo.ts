import Owner from './Owner';

/**
 * Represents a GitHub repository's information.
 *
 * This interface defines the structure of information associated with a GitHub repository.
 * It includes details such as the repository's name, owner, whether it's private, and more.
 *
 * @interface Repo
 * @example
 * ```ts
 * const repository: Repo = {
 *   id: 12345,
 *   name: "example-repo",
 *   full_name: "username/example-repo",
 *   private: false,
 *   owner: {
 *     login: "exampleOwner",
 *     id: 123,
 *     // ...owner properties
 *   },
 *   // ...other properties
 * };
 * ```
 */
export default interface Repo {
  id: number;
  node_id: string;
  name: string;
  full_name: string;
  private: boolean;
  owner?: Owner;
}
