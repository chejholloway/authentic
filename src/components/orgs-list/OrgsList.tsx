import React from "react";
import { Link } from "react-router-dom";
import {
  List,
  ListItem,
  ListItemPrefix,
  Avatar,
  Card,
  Typography,
} from "@material-tailwind/react";

import Org from "../../types/Org";
import Developer from "../../types/Developer";

interface OrgsListProps {
  developer: Developer | undefined;
  org: Org[] | undefined;
}

const OrgsList: React.FC<OrgsListProps> = ({ org }) => {
  return (
    <div className="text-gray-700">
      <h3 className="text-center">
        <Typography className="font-normal">Organizations</Typography>
      </h3>
      {org ? (
        org.map((org) => (
          <div key={org.id}>
            <Link to={"/"}>
              <Card className="w-full">
                <List key={org.id}>
                  <ListItem>
                    <ListItemPrefix>
                      <Avatar
                        variant="circular"
                        alt={org?.login}
                        src={org?.avatar_url}
                      />
                    </ListItemPrefix>
                    <Typography
                      variant="small"
                      color="gray"
                      className="font-normal"
                    >
                      {org.login}
                    </Typography>
                  </ListItem>
                </List>
              </Card>
            </Link>
          </div>
        ))
      ) : (
        <p>No organizations available.</p>
      )}
    </div>
  );
};

export default OrgsList;
