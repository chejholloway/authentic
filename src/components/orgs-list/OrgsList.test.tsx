import { render } from '@testing-library/react';
import React from 'react';
import { Link } from "react-router-dom";
import {
  List,
  ListItem,
  ListItemPrefix,
  Avatar,
  Card,
  Typography,
} from "@material-tailwind/react";
import Org from "../../types/Org";
import Developer from "../../types/Developer";
import Repo from '../../types/Repo';
import OrgsList from "./OrgsList";

jest.mock("react-router-dom");
jest.mock("@material-tailwind/react");
jest.mock("../../types/Org");
jest.mock('../../types/Repo');
jest.mock("../../types/Developer");

const mockOrgs: Org[] = [
  {
    "avatar_url": "https://avatars.githubusercontent.com/u/28929274?v=4",
    "description": "Plataforma de educaÃ§Ã£o em tecnologia ðŸš€",
    "events_url": "https://api.github.com/users/charlenopires/events{/privacy}",
    "hooks_url": "https://api.github.com/orgs/Rocketseat/hooks",
    "id": 28929273,
    "issues_url": "https://api.github.com/orgs/Rocketseat/issues",
    "login": "Rocketseat",
    "members_url": "https://api.github.com/orgs/Rocketseat/members{/member}",
    "node_id": "MDEyOk9yZ2FuaXphdGlvbjI4OTI5Mjc0",
    "public_members_url": "https://api.github.com/orgs/Rocketseat/public_members{/member}",
    "repos_url": "https://api.github.com/orgs/Rocketseat/repos",
    "url": "https://api.github.com/orgs/Rocketseat",
  },
  {
    "avatar_url": "https://avatars.githubusercontent.com/u/28929274?v=4",
    "description": "Plataforma de educaÃ§Ã£o em tecnologia ðŸš€",
    "events_url": "https://api.github.com/users/charlenopires/events{/privacy}",
    "hooks_url": "https://api.github.com/orgs/Rocketseat/hooks",
    "id": 28929274,
    "issues_url": "https://api.github.com/orgs/Rocketseat/issues",
    "login": "Rocketseat",
    "members_url": "https://api.github.com/orgs/Rocketseat/members{/member}",
    "node_id": "MDEyOk9yZ2FuaXphdGlvbjI4OTI5Mjc0",
    "public_members_url": "https://api.github.com/orgs/Rocketseat/public_members{/member}",
    "repos_url": "https://api.github.com/orgs/Rocketseat/repos",
    "url": "https://api.github.com/orgs/Rocketseat",
  },
];

const mockDeveloper: Developer = {
  "login": "ruanyf",
  "avatar_url": "https://api.github.com/users/ruanyf/starred{/owner}{/repo}",
};

describe('<OrgsList>', () => {
  it('should render component', () => {
    const { container } = render(
      <OrgsList developer={mockDeveloper} orgs={mockOrgs}/>
    );

    expect(container).toMatchSnapshot();
  });
});
