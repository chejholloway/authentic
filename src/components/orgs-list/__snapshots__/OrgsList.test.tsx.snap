// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrgsList> should render component 1`] = `
<div>
  <div
    class="text-gray-700"
  >
    <h3
      class="text-center"
    >
      <h1
        class="font-normal css-qhut37"
      >
        Organizations
      </h1>
    </h3>
    <div />
    <div />
  </div>
</div>
`;
